src/viz-parole-revocation/VizParoleRevocationContainer.js

```diff
@@ -0,0 +1,92 @@
+import PropTypes from "prop-types";
+import { ascending, group } from "d3-array";
+import React, { useState, useEffect } from "react";
+import VizParoleRevocation from "./VizParoleRevocation";
+import {
+  DIMENSION_KEYS,
+  TOTAL_KEY,
+  VIOLATION_TYPES,
+  THEME,
+} from "../constants";
+
+// TODO: may not be necessary if source files change
+function typeCastRecords(records) {
+  return records.map((record) => ({
+    ...record,
+    year: +record.year,
+    month: +record.month,
+    revocation_count: +record.revocation_count,
+  }));
+}
+
+function groupByMonth(records) {
+  // order chronologically for convenience
+  const sorted = typeCastRecords([...records]).sort(
+    (a, b) => ascending(a.year, b.year) || ascending(a.month, b.month)
+  );
+  return group(sorted, ({ month, year }) => `${year}-${month}`);
+}
+
+const VIOLATION_REASONS_COLORS = THEME.colors.violationReasons;
+
+export default function VizParoleRevocationContainer({
+  data: { paroleRevocationByMonth },
+  month,
+  setMonthList,
+  dimension,
+  ...passThruProps
+}) {
+  const [monthlyData] = useState(groupByMonth(paroleRevocationByMonth));
+
+  useEffect(() => {
+    setMonthList(Array.from(monthlyData.keys()));
+  }, [monthlyData, setMonthList]);
+
+  // TODO: loading state
```

Comment:
This TODO is going to be covered only _later_ when we hook up to the API, right? Not now?

Reply:
yeah, i don't think i should leave this one in either